homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+        self._name = self._github_data.name
+        self._stargazers = self._github_data.stargazers
+        self._topics = self._github_data.topics
+        self._branches = self._github_data.branches
+        self._open_issues = self._github_data.open_issues
+        self._open_pull_requests = self._github_data.open_pull_requests
+        self._last_commit = self._github_data.last_commit
+        self._latest_release = self._github_data.latest_release
+        self._available = self._github_data.available
+
+
+class GitHubData():
+    """GitHub Data object."""
+
+    def __init__(self, interval, repository, access_token=None, username=None,
+                 password=None, server_url=None):
+        """Set up GitHub."""
+        import github
+
+        self._github = github
+
+        self.setup_error = False
+
+        try:
+            if server_url is not None:
+                server_url += "/api/v3"
+                if access_token is not None:
+                    self._github_obj = github.Github(
+                        access_token, base_url=server_url)
+                elif username is not None and password is not None:
+                    self._github_obj = github.Github(
+                        username, password, base_url=server_url)
+            else:
+                if access_token is not None:
+                    self._github_obj = github.Github(access_token)
+                elif username is not None and password is not None:
+                    self._github_obj = github.Github(username, password)
+
+            self.repository_path = repository[CONF_PATH]
+
+            repo = self._github_obj.get_repo(self.repository_path)
+        except self._github.BadCredentialsException as err:
+            _LOGGER.error("Bad Credentials for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.UnknownObjectException as err:
+            _LOGGER.error("UnknownObjectException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.RateLimitExceededException as err:
+            _LOGGER.error("RateLimitExceededException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.BadAttributeException as err:
+            _LOGGER.error("BadAttributeException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.TwoFactorException as err:
+            _LOGGER.error("TwoFactorException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.GithubException as err:
+            _LOGGER.error("GithubException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
```

Comment:
This is very verbose. Is there a `github` base exception that we could just catch?